app/controllers/users/approvals_controller.rb

```diff
@@ -1,10 +1,14 @@
 class Users::ApprovalsController < ApplicationController
 
-  before_action :authenticate_user!
+  before_action :authenticate_user! 
+  before_action :check_user, only: :index
 
   def index
     @approved_devs = current_user.approved_developers
     @pending_approvals = current_user.pending_approvals
+    if @pending_approvals.length == 0 && params[:redirect]
+      redirect_to params[:redirect]
```

Comment:
I like this solution, although it will make it harder for the controller to decide between redirecting locally and redirecting to the developer.redirect_path but I guess i can pass something safer in the params to indicate this.